### PR TITLE
Cloud Migration: Remove ID migration in favor of UID

### PR DESCRIPTION
**Pull request by leandro-deveikis**
**What is this feature?**
Replaces how we identify entities for Cloud Migration service. Before this we used IDs (integer auto incremental). With this PR, this changes to use UIDs as the rest of the grafana code. 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-operator-experience-squad/issues/721

**Special notes for your reviewer:**
Details:
- Added migration to add a new column UID to both migration and migration_run tables.
- Added migration to set UID for already existing records in the database before adding a new column.
- Added UID field to CloudMigration and CloudMigrationRun dtos (model.go)
- Changed APIs to use UID instead of IDs. Updated OpenApi specs. 
- Update frontend code to adjust to the changes. 
- Make backend interfaces consistent
- Remove {uid} parameter from the fetch run endpoint

### CODE REVIEW

File: pkg/services/cloudmigration/api/api.go [hunk 1]
```diff
@@ -202,10 +195,14 @@ type RunCloudMigrationRequest struct {
 // 403: forbiddenError
 // 500: internalServerError
 func (cma *CloudMigrationAPI) GetMigrationRun(c *contextmodel.ReqContext) response.Response {
+	// TODO Parameter 'uid' is not actually used at the moment, since we just need the run UID.
```
Comment (leandro-deveikis) [inline pkg/services/cloudmigration/api/api.go:198]: check comment in PR description about this

File: pkg/services/sqlstore/migrations/cloud_migrations.go [hunk 2]
```diff
@@ -37,4 +37,30 @@ func addCloudMigrationsMigrations(mg *Migrator) {
...
 	mg.AddMigration("add region_slug column", NewAddColumnMigration(migrationTable, &regionSlugColumn))
 	mg.AddMigration("add cluster_slug column", NewAddColumnMigration(migrationTable, &clusterSlugColumn))
+
+	// --- adding uid to migration
+	migUidColumn := Column{Name: "uid", Type: DB_NVarchar, Length: 40, Nullable: true}
+	mg.AddMigration("add migration uid column", NewAddColumnMigration(migrationTable, &migUidColumn))
+
+	mg.AddMigration("Update uid column values for migration", NewRawSQLMigration("").
+		SQLite("UPDATE cloud_migration SET uid=printf('u%09d',id) WHERE uid IS NULL;").
+		Postgres("UPDATE `cloud_migration` SET uid='u' || lpad('' || id::text,9,'0') WHERE uid IS NULL;").
```
Comment (leandro-deveikis) [inline pkg/services/sqlstore/migrations/cloud_migrations.go:47]: test with postgres

File: pkg/services/cloudmigration/api/api.go (hunk 1 above)
Comment (mmandrus) [reply pkg/services/cloudmigration/api/api.go]: Decided to take care of it here so we can stop messing with this :D

File: pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go [hunk 3]
```diff
@@ -41,6 +42,9 @@ func (ss *sqlStore) GetMigration(ctx context.Context, id int64) (*cloudmigration
 
 func (ss *sqlStore) SaveMigrationRun(ctx context.Context, cmr *cloudmigration.CloudMigrationRun) error {
 	return ss.db.WithDbSession(ctx, func(sess *db.Session) error {
+		if cmr.UID == "" {
+			cmr.UID = util.GenerateShortUID()
+		}
```
[the remark this answers is not in the document]
Comment (mmandrus) [reply pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go]: no longer applies, have I have explicitly made this a "create" function that sets the UID no matter what

**Review by joshhunt — COMMENTED** [review]
✅ Frontend looks good. Be sure to run `git add public/app/features/migrate-to-cloud/api/endpoints.gen.ts` after changing the API spec to ensure the right frontend code is generated. I've pushed up the latest changes to it.

**Review by PoorlyDefinedBehaviour — APPROVED** [review]

**Pull request merged**